src/atoms/second_order_cone/rationalnorm.jl

```diff
@@ -63,69 +63,68 @@ end
-                  "more accurate solutions")
-    end
-    # u corresponds to "introduced" variables; make a matrix of them
-    # and then add equality constraints for the first and second
-    # variable in the matrix (i.e. first and second columns), as they
-    # correspond to s and v.
-    u = Variable(length(var_list), d);
-    # v is the first variable
-    conic_form!(v == u[var_list[1], :]', unique_conic_forms)
-    # s[jj] is the second variable
-    conic_form!(s == u[var_list[2], :]', unique_conic_forms)
+        # Reduce to SOC constraints (get powers for each element)
+        num = Int(numerator(x.k));
+        denom = Int(denominator(x.k));
+        # Construct list of inequalities of form u^2 <= vw, where the list
+        # is given by triples in ineq_list.
+        (ineq_list,
+         var_list) = psocp.ProductToSimpleInequalities(denom, num - denom);
+        if (length(ineq_list) > 10)
+            @warn string("Rational norm generating ", length(ineq_list),
+                         " intermediate constraints.\n\tIncreasing ",
+                         ":max_iters or decreasing solver tolerance\n\tmay give ",
+                         "more accurate solutions")
+        end
+        # u corresponds to "introduced" variables; make a matrix of them
+        # and then add equality constraints for the first and second
+        # variable in the matrix (i.e. first and second columns), as they
+        # correspond to s and v.
+        u = Variable(length(var_list), d);
+        # v is the first variable
+        conic_form!(v == u[var_list[1], :]', unique_conic_forms)
+        # s[jj] is the second variable
+        conic_form!(s == u[var_list[2], :]', unique_conic_forms)
 
-    for jj = 1:d
-      # t is the third variable
-      conic_form!(t == u[var_list[3], jj], unique_conic_forms)
-      for ii = 1:length(ineq_list)
-        temp1 = u[ineq_list[ii].left_var_ind, jj];
-        temp2 = u[ineq_list[ii].t_var_ind, jj];
-        temp3 = u[ineq_list[ii].s_var_ind, jj];
-        # Want: t1^2 <= t2 * t3, so add constraints of form
-        # norm([2 * t1, t2 - t3], 2) <= t2 + t3.
-        conic_form!(SOCElemConstraint(temp2 + temp3,
-                                      temp2 - temp3, 2 * temp1),
-                    unique_conic_forms);
-      end
+        for jj = 1:d
+            # t is the third variable
+            conic_form!(t == u[var_list[3], jj], unique_conic_forms)
+            for ii = 1:length(ineq_list)
+                temp1 = u[ineq_list[ii].left_var_ind, jj];
+                temp2 = u[ineq_list[ii].t_var_ind, jj];
+                temp3 = u[ineq_list[ii].s_var_ind, jj];
+                # Want: t1^2 <= t2 * t3, so add constraints of form
+                # norm([2 * t1, t2 - t3], 2) <= t2 + t3.
+                conic_form!(SOCElemConstraint(temp2 + temp3,
+                                              temp2 - temp3, 2 * temp1),
+                            unique_conic_forms);
```

Comment:
🤢

Reply:
Preexisting condition (also tbh this is likely how I'd write it in my own project...)